src/DependencyInjection/SonataFormatterExtension.php

```diff
@@ -152,16 +149,10 @@ private function createEnvironment(
 
         $container->setDefinition(sprintf('sonata.formatter.twig.sandbox.%s', $code), $sandbox);
 
-        $formatterDefinition = $container->getDefinition($formatterId);
-        if (
-            null !== $formatterDefinition->getClass()
-            && is_a($formatterDefinition->getClass(), ExtendableFormatter::class, true)
-        ) {
-            $formatterDefinition->addMethodCall('addExtension', [new Reference(sprintf('sonata.formatter.twig.sandbox.%s', $code))]);
+        $env->addMethodCall('addExtension', array(new Reference(sprintf('sonata.formatter.twig.sandbox.%s', $code))));
 
-            foreach ($extensions as $extension) {
-                $formatterDefinition->addMethodCall('addExtension', [new Reference($extension)]);
-            }
+        foreach ($extensions as $extension) {
+            $env->addMethodCall('addExtension', array(new Reference($extension)));
```

Comment:
This was changed like 2-3 times, from a working code to a non working code...

https://github.com/sonata-project/SonataFormatterBundle/pull/664
https://github.com/sonata-project/SonataFormatterBundle/pull/417

cc @VincentLanglet

Reply:
Since 2019, adding extensions to formatters does not work.

Reply:
Yes this was reported. I thought I fixed this :s